features/step_definitions/kata.rb

```diff
@@ -1,229 +1,127 @@
+      @result_state = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[5].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_state
     else
-      step %Q/I switch to cluster admin pseudo user/
-      step %Q|I obtain test data file "kata/release-#{cb.master_version}/deployment.yaml"|
-      @result = user.cli_exec(:apply, f: "deployment.yaml")
-      raise "Failed to deploy kata operator" unless @result[:success]
-      project(kata_ns)
+      @result_status = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[7].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_status
+      @result_state = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[8].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_state
     end
-
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-    step %Q|I obtain test data file "kata/release-#{cb.master_version}/kataconfiguration_v1_kataconfig.yaml"|
-    @result = user.cli_exec(:apply, f: 'kataconfiguration_v1_kataconfig.yaml')
-    raise "Failed to apply kataconfig" unless @result[:success]
-    step %Q/I wait for "#{kata_config_name}" install to start/
-  else
-    logger.info("There's already an existing 'kataconfig' resuing it...")
-    project(kata_ns)
-    step %Q/I switch to cluster admin pseudo user/
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
   end
-  logger.info("Using kata image: #{pod.container_specs.first.image}")
-  step %Q/I wait until number of completed kata runtime nodes match for "#{kata_config_name}"/
+  raise "Failed to install sandboxed operator" unless @result_state
+  logger.info("Sandboxed operator installation status is #{expected_state}")
+end
+
+When /^I apply #{QUOTED} in #{QUOTED} namespace$/ do |kata_config_name, kata_ns|
+  step %Q/I store master major version in the :master_version clipboard/
+  step %Q|I obtain test data file "kata/release-#{cb.master_version}/kataconfiguration_v1_kataconfig.yaml"|
+  @result = user.cli_exec(:apply, f: 'kataconfiguration_v1_kataconfig.yaml', n:kata_ns)
+  raise "Failed to apply #{kata_config_name} kataconfig" unless @result[:success]
+end
+
+Then /^Kata runtime installed on selected worker nodes for #{QUOTED} kataconfig$/ do |kataconfig_name|
+  step %Q|I wait until number of completed kata runtime nodes for "#{kataconfig_name}" matches|
+end
+
+When /^I apply #{QUOTED} pod in #{QUOTED} namespace$/ do |pod_name, kata_ns|
+  step %Q|I obtain test data file "kata/#{pod_name}.yaml"|
+  @result = user.cli_exec(:apply, f: "#{pod_name}.yaml", n:kata_ns)
+  raise "Failed to deploy a pod" unless @result[:success]
```

Comment:
@valtair1891 this error message is misleading, since `oc apply` does not check a pod is deployed.